spec/models/scanned_resource_spec.rb

```diff
@@ -174,4 +174,33 @@
       expect(subject.valid?).to eq false
     end
   end
+
+  describe "#check_completion" do
+    it "completes record when state changes to 'complete'" do
+      allow(subject).to receive("state_changed?").and_return true
+      subject.state = 'complete'
+      expect(subject).to receive(:complete_record)
+      expect { subject.check_completion }.to change { ActionMailer::Base.deliveries.count }.by(1)
```

Comment:
Left a note in Slack, but for completion: The problem is the mock on :complete_record above. It replaces that method so it doesn't call anything. Change to something like:

```ruby
  describe "#check_completion", vcr: { cassette_name: "ezid" } do
    it "completes record when state changes to 'complete'" do
      allow(subject).to receive("state_changed?").and_return true
      allow(subject).to receive(:complete_record).and_call_original
      subject.state = 'complete'
      expect { subject.check_completion }.to change { ActionMailer::Base.deliveries.count }.by(1)
      expect(subject).to have_received(:complete_record)
    end
```